### ISSUE TITLE
Ubuntu xenial: psycopgmodule.c:426: undefined reference to `_Py_NoneStruct'

### ISSUE THREAD

**Issue by df7cb** [psycopg/psycopg2#963]
Trying to build (the Debian package of) 2.8.3 on Ubuntu xenial yields this:
```
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fno-strict-aliasing -g -O2 -fPIE -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -DHAVE_MXDATETIME=1 -DPSYCOPG_VERSION=2.8.3 (dt dec mx pq3 ext lo64) -DPG_VERSION_NUM=110005 -DHAVE_LO64=1 -I/usr/include/python2.7/mx -I/usr/include/python2.7 -I. -I/usr/include/postgresql -I/usr/include/postgresql/11/server -c psycopg/typecast.c -o build/temp.linux-x86_64-2.7/psycopg/typecast.o -Wdeclaration-after-statement
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fno-strict-aliasing -g -O2 -fPIE -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -DHAVE_MXDATETIME=1 -DPSYCOPG_VERSION=2.8.3 (dt dec mx pq3 ext lo64) -DPG_VERSION_NUM=110005 -DHAVE_LO64=1 -I/usr/include/python2.7/mx -I/usr/include/python2.7 -I. -I/usr/include/postgresql -I/usr/include/postgresql/11/server -c psycopg/adapter_mxdatetime.c -o build/temp.linux-x86_64-2.7/psycopg/adapter_mxdatetime.o -Wdeclaration-after-statement
x86_64-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -Wl,-z,relro -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -Wdate-time -D_FORTIFY_SOURCE=2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wl,-Bsymbolic-functions -fPIE -pie -Wl,-z,relro -Wl,-z,now -g -O2 -fPIE -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 build/temp.linux-x86_64-2.7/psycopg/psycopgmodule.o build/temp.linux-x86_64-2.7/psycopg/green.o build/temp.linux-x86_64-2.7/psycopg/pqpath.o build/temp.linux-x86_64-2.7/psycopg/utils.o build/temp.linux-x86_64-2.7/psycopg/bytes_format.o build/temp.linux-x86_64-2.7/psycopg/libpq_support.o build/temp.linux-x86_64-2.7/psycopg/win32_support.o build/temp.linux-x86_64-2.7/psycopg/solaris_support.o build/temp.linux-x86_64-2.7/psycopg/connection_int.o build/temp.linux-x86_64-2.7/psycopg/connection_type.o build/temp.linux-x86_64-2.7/psycopg/cursor_int.o build/temp.linux-x86_64-2.7/psycopg/cursor_type.o build/temp.linux-x86_64-2.7/psycopg/column_type.o build/temp.linux-x86_64-2.7/psycopg/replication_connection_type.o build/temp.linux-x86_64-2.7/psycopg/replication_cursor_type.o build/temp.linux-x86_64-2.7/psycopg/replication_message_type.o build/temp.linux-x86_64-2.7/psycopg/diagnostics_type.o build/temp.linux-x86_64-2.7/psycopg/error_type.o build/temp.linux-x86_64-2.7/psycopg/conninfo_type.o build/temp.linux-x86_64-2.7/psycopg/lobject_int.o build/temp.linux-x86_64-2.7/psycopg/lobject_type.o build/temp.linux-x86_64-2.7/psycopg/notify_type.o build/temp.linux-x86_64-2.7/psycopg/xid_type.o build/temp.linux-x86_64-2.7/psycopg/adapter_asis.o build/temp.linux-x86_64-2.7/psycopg/adapter_binary.o build/temp.linux-x86_64-2.7/psycopg/adapter_datetime.o build/temp.linux-x86_64-2.7/psycopg/adapter_list.o build/temp.linux-x86_64-2.7/psycopg/adapter_pboolean.o build/temp.linux-x86_64-2.7/psycopg/adapter_pdecimal.o build/temp.linux-x86_64-2.7/psycopg/adapter_pint.o build/temp.linux-x86_64-2.7/psycopg/adapter_pfloat.o build/temp.linux-x86_64-2.7/psycopg/adapter_qstring.o build/temp.linux-x86_64-2.7/psycopg/microprotocols.o build/temp.linux-x86_64-2.7/psycopg/microprotocols_proto.o build/temp.linux-x86_64-2.7/psycopg/typecast.o build/temp.linux-x86_64-2.7/psycopg/adapter_mxdatetime.o -L/usr/lib/x86_64-linux-gnu -lpq -o /home/cbe/projects/postgresql/psycopg2/psycopg2/.pybuild/pythonX.Y_2.7/build/psycopg2/_psycopg.so
/usr/lib/gcc/x86_64-linux-gnu/5/../../../x86_64-linux-gnu/Scrt1.o: In function `_start':
(.text+0x20): undefined reference to `main'
build/temp.linux-x86_64-2.7/psycopg/psycopgmodule.o: In function `encrypt_password':
/home/cbe/projects/postgresql/psycopg2/psycopg2/psycopg/psycopgmodule.c:426: undefined reference to `_Py_NoneStruct'
/home/cbe/projects/postgresql/psycopg2/psycopg2/psycopg/psycopgmodule.c:432: undefined reference to `_PyArg_ParseTupleAndKeywords_SizeT'
/home/cbe/projects/postgresql/psycopg2/psycopg2/psycopg/psycopgmodule.c:443: undefined reference to `PyType_IsSubtype'
/home/cbe/projects/postgresql/psycopg2/psycopg2/psycopg/psycopgmodule.c:446: undefined reference to `PyType_IsSubtype'
/home/cbe/projects/postgresql/psycopg2/psycopg2/psycopg/psycopgmodule.c:495: undefined reference to `PyString_FromString'
/home/cbe/projects/postgresql/psycopg2/psycopg2/psycopg/psycopgmodule.c:450: undefined reference to `PyExc_TypeError'
/home/cbe/projects/postgresql/psycopg2/psycopg2/psycopg/psycopgmodule.c:450: undefined reference to `PyErr_SetString'
/home/cbe/projects/postgresql/psycopg2/psycopg2/psycopg/psycopgmodule.c:499: undefined reference to `PyErr_Format'
/home/cbe/projects/postgresql/psycopg2/psycopg2/psycopg/psycopgmodule.c:476: undefined reference to `PyErr_SetString'
build/temp.linux-x86_64-2.7/psycopg/psycopgmodule.o: In function `register_type':
/home/cbe/projects/postgresql/psycopg2/psycopg2/psycopg/psycopgmodule.c:240: undefined reference to `_PyArg_ParseTuple_SizeT'
/home/cbe/projects/postgresql/psycopg2/psycopg2/psycopg/psycopgmodule.c:244: undefined reference to `_Py_NoneStruct'
/home/cbe/projects/postgresql/psycopg2/psycopg2/psycopg/psycopgmodule.c:245: undefined reference to `PyType_IsSubtype'
/home/cbe/projects/postgresql/psycopg2/psycopg2/psycopg/psycopgmodule.c:252: undefined reference to `PyType_IsSubtype'
/home/cbe/projects/postgresql/psycopg2/psycopg2/psycopg/psycopgmodule.c:248: undefined reference to `PyDict_New'
/home/cbe/projects/postgresql/psycopg2/psycopg2/psycopg/psycopgmodule.c:258: undefined reference to `PyExc_TypeError'
/home/cbe/projects/postgresql/psycopg2/psycopg2/psycopg/psycopgmodule.c:258: undefined reference to `PyErr_SetString'
[...]
collect2: error: ld returned 1 exit status
error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
E: pybuild pybuild:274: build: plugin distutils failed with: exit code=1: /usr/bin/python setup.py build 
dh_auto_build: pybuild --build -i python{version} -p 2.7 --dir . returned exit code 13
debian/rules:12: recipe for target 'override_dh_auto_build' failed
```
Full log: https://pgdgbuild.dus.dg-i.net/job/psycopg2-binaries/6/architecture=amd64,distribution=xenial/consoleFull

It works fine on Ubuntu bionic (and Debian stretch/buster/unstable).

Is that a psycopg2 bug, or a general problem with xenial?

Context is that pgadmin4 recently moved to require psycopg2 2.8.x, and I'm looking into upgrading the psycopg2 package on apt.postgresql.org.

**Comment by dvarrazzo** [comment]
We don't maintain the debian packages. The only way I support building the package is what is documented in the docs. We only make sure that `python setup.py build` and/or `make` work; I don't have knowledge about other build tools I'm afraid.

**Comment by dvarrazzo** [comment]
Speaking of packaging, I would like to release 2.8.4 soon, which should include an important fix (#943) and a pgadmin-related fix (#961) so maybe it would be good to coordinate :)

Thank you very much for your work!

**Comment by df7cb** [comment]
I just tried manually building on xenial and that works indeed. Sorry for the noise, will have to dive into the python stack there then...
